lib/activity/router.js

```diff
@@ -76,6 +76,38 @@ module.exports = function route(callback) {
         // Delay GitHub API calls to avoid replication lag
         context.github.hook.before('request', avoidReplicationLag());
 
+        // Label filtering
+        const issue = context.payload.issue || context.payload.pull_request;
+        const shouldFilterByLabel = (() => {
+          if (issue === undefined) {
+            return false;
+          }
+
+          if (!Array.isArray(issue.labels)) {
+            return false;
+          }
+
+          if (subscription.settings.label === undefined) {
```

Comment:
```suggestion
          if (subscription.settings && subscription.settings.label === undefined) {
```

Is there guarantee that `subscription.settings` is available at this point? If it's not then you might get `Uncaught TypeError: Cannot read property of undefined`